web3/utils/blocks.py

```diff
@@ -1,10 +1,44 @@
 from eth_utils import (
+    is_hex,
     is_string,
+    is_integer,
+    remove_0x_prefix,
     force_text,
 )
 
 
-def is_predefined_block_number(block_number):
-    if not is_string(block_number):
+def is_predefined_block_number(value):
+    if not is_string(value):
         return False
-    return force_text(block_number) in {"latest", "pending", "earliest"}
+    return force_text(value) in {"latest", "pending", "earliest"}
+
+
+def is_hex_encoded_block_hash(value):
+    if not is_string(value):
+        return False
+    return len(remove_0x_prefix(value)) == 64 and is_hex(value)
+
+
+def is_hex_encoded_block_number(value):
+    if not is_string(value):
+        return False
+    elif is_hex_encoded_block_hash(value):
+        return False
+    try:
+        value_as_int = int(value, 16)
+    except ValueError:
+        return False
+    return 0 <= value_as_int < 2**256 - 1
```

Comment:
off by one, it looks like: `0 <= value_as_int < 2**256`

Reply:
👍